### ISSUE TITLE
All cards - Error Prevention - No warning message to prevent the user accidentally deleting a record

### ISSUE THREAD

**Issue by khodgkinson-he** [archesproject/arches#9091]
**Arches Version/Branch:** dev/6.1.x 

**Describe the bug and how to reproduce it:**
(HE Notes : Although this has come up as an accessibility issue, it should go back as a core giveback feature due to the usability improvement and there should be no breaking changes by giving this straight back.)

**Steps to Reproduce:**
Navigate to any page in Keystone containing cards (e.g. System Settings)
Attempt to delete any card.

**Expected Result:**
User should be presented with a warning message on pressing the 'Delete' button, asking them if they are sure they want to delete the record they have selected.

**Actual Result:**
No warning message on the user selecting 'Delete' button to prevent the user accidentally deleting a record.
